datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -781,14 +790,18 @@ async def dimension_join_path(
     list of dimension links that represent the join path
     """
     # Check if it is a local dimension
-    if dimension.startswith(node.name):
-        for col in node.columns:  # pragma: no cover
-            # Decide if we should restrict this to only columns marked as dimensional
-            # await session.refresh(col, ["attributes"]) TODO
-            # if col.is_dimensional():
-            #     ...
-            if f"{node.name}.{col.name}" == dimension:
-                return []
+    for col in node.columns:  # pragma: no cover
+        # Decide if we should restrict this to only columns marked as dimensional
```

Comment:
I'm ok with restricting this only to columns marked as dimensional. It's reasonable that users will want some control of this and it's easy enough to mark a ton of columns as dimensional if that's what they want.

Reply:
This is a similar conundrum to my question about the PK dim columns. I feel like we should start with more limitation early on (to keep things clean) and open up later if needed. Going to other way may be harder.

Reply:
Yeah, I agree, I'll do this in a follow-up PR though, since this change wasn't actually about this problem.